src/lu.jl

```diff
@@ -1,14 +1,27 @@
+# define our own LU type, since LinearAlgebra.LU requires p::Vector
+struct LU{L,U,p}
+    L::L
+    U::U
+    p::p
+end
```

Comment:
`L::L`, `U::U`, `p::p` seems like it could get confusing.

Reply:
Interesting problem. This normally is avoided by the fact that we use upper case types and lower case fields. `LinearAlgebra` is particularly *nasty* at ignoring this advice, leading to this kind of... stuff.

The interface for this factorization from `LinearAlgebra` seems to involve defining properties `L` and `U`, so we can't ignore that. The type parameters for the type needn't suffer terribly either - do we really want `LU{LT, UT, pT}`? `LU{L_t, U_t, p_t}`? Perhaps what's here currently is the least of all evils.

Reply:
OTOH our `SVD` type already uses ugly type parameters... ¯\_(ツ)_/¯

Reply:
Yeah, it's probably alright. It was surprising to me at least since I hadn't seen this pattern before, but I guess you're right that the param names won't leak out to user code.

Reply:
Yea, I always assumed this was not valid, and usually use this pattern if needed
```
struct A{TT}
    T::TT
end
```